runtime/runtime.m

```diff
@@ -1051,7 +1051,7 @@ -(void) xamarinSetFlags: (enum XamarinGCHandleFlags) flags;
 void
 xamarin_add_registration_map (struct MTRegistrationMap *map, bool partial)
 {
-	if (!strcmp (map->product_hash, PRODUCT_HASH)) {
+	if (strcmp (map->product_hash, PRODUCT_HASH) != 0) {
```

Comment:
Woudln't strncmp be a better choice here? I'm pretty sure `strcmp` doesn't have to do `strlen` first, but just making sure.

Reply:
I don't think there's any difference.